cli/cmd/project/refresh.go

```diff
@@ -26,7 +26,7 @@ func RefreshCmd(ch *cmdutil.Helper) *cobra.Command {
 			if len(args) > 0 {
 				project = args[0]
 			}
-			if !local && !cmd.Flags().Changed("project") && len(args) == 0 && ch.Interactive {
+			if !local && !cmd.Flags().Changed("project") && len(args) == 0 && ch.Interactive && cmd.Flags().NFlag() == 0 {
 				var err error
 				project, err = ch.InferProjectName(cmd.Context(), ch.Org, path)
```

Comment:
Same comment as above, I think. In this case, would the following perhaps be reasonable?
```suggestion
			if !local && project == "" {
				if !ch.Interactive {
					return fmt.Errorf("project not specified and could not be inferred from context")
				}

				var err error
				project, err = ch.InferProjectName(cmd.Context(), ch.Org, path)
				...
```